client/data/media/use-add-external-media.js

```diff
@@ -0,0 +1,31 @@
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { useCallback } from 'react';
+import wpcom from 'calypso/lib/wp';
+import { useUploadMediaMutation } from './use-upload-media-mutation';
+
+const getExternalUploader = ( service ) => ( file, siteId, postId = 0 ) =>
+	wpcom.req.post( `/sites/${ siteId }/external-media-upload`, {
+		external_ids: [ file.guid ],
+		service,
+		post_id: postId,
+	} );
+
+export const useAddExternalMedia = () => {
+	const { uploadMediaAsync } = useUploadMediaMutation();
+
+	return useCallback(
+		( file, site, postId, service ) => {
```

Comment:
I think some docs (inline or not) would be great for those. Right now a developer needs to read this entire hook to understand how to use it. We could reuse the inline docs of the removed corresponding thunk.